### PR TITLE
refactor: Port all direct CodedString usages to StyledText

### PR DESCRIPTION
**Pull request by Mcrtin**
Ports the remaining usages of CodedString to StyledText.

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java [hunk 1]
```diff
@@ -146,14 +145,13 @@ private void tryAutoSearch(AbstractContainerScreen<?> abstractContainerScreen) {
...
-        CodedString name = ComponentUtils.getCoded(abstractContainerScreen
+        StyledText name = StyledText.fromComponent(abstractContainerScreen
                 .getMenu()
                 .getItems()
                 .get(currentSearchableContainerType.getNextItemSlot())
                 .getHoverName());
 
-        if (!name.getMatcher(currentSearchableContainerType.getNextItemPattern())
-                .matches()) {
+        if (!name.matches(currentSearchableContainerType.getNextItemPattern())) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java:154]: Did you check all patterns here?

File: common/src/main/java/com/wynntils/features/inventory/ItemScreenshotFeature.java [hunk 2]
```diff
@@ -125,10 +125,10 @@ private void takeScreenshot(Screen screen, Slot hoveredSlot, List<Component> ite
 
         if (saveToDisk.get()) {
             // First try to save it to disk
-            String itemNameForFile = WynnUtils.normalizeBadString(ComponentUtils.stripFormatting(
-                            CodedString.fromString(itemStack.getHoverName().getString())))
+            String itemNameForFile = WynnUtils.normalizeBadString(StyledText.fromComponent(itemStack.getHoverName())
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/inventory/ItemScreenshotFeature.java]: Pretty sure you can do StyledText.fromComponent(..).normalized().getString() instead

File: common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java [hunk 3]
```diff
@@ -57,14 +55,18 @@ public void add(
...
+            Matcher matcher = null;
+
+            for (TrackedBar bar : knownBars) {
+                matcher = StyledText.fromComponent(name).getMatcher(bar.pattern);
+                if (matcher.matches()) {
+                    trackedBar = bar;
+                    break;
+                }
+            }
+            if (trackedBar == null) return;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java]: Why was this changed?

File: common/src/main/java/com/wynntils/features/inventory/ItemScreenshotFeature.java (hunk 2 above)
Comment (Mcrtin) [reply common/src/main/java/com/wynntils/features/inventory/ItemScreenshotFeature.java]: I didn't know they were doing the same thing

File: common/src/main/java/com/wynntils/handlers/scoreboard/type/ScoreboardLineChange.java [hunk 4]
```diff
@@ -4,7 +4,7 @@
  */
 package com.wynntils.handlers.scoreboard.type;
 
-import com.wynntils.core.text.CodedString;
+import com.wynntils.core.text.StyledText;
 import net.minecraft.server.ServerScoreboard;
 
-public record ScoreboardLineChange(CodedString lineText, ServerScoreboard.Method method, int lineIndex) {}
+public record ScoreboardLineChange(StyledText lineText, ServerScoreboard.Method method, int lineIndex) {}
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/handlers/scoreboard/type/ScoreboardLineChange.java]: How come this wasn't changed in the scoreboard PR and it still works?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/handlers/scoreboard/type/ScoreboardLineChange.java]: Oh so this file is not used. Can you just delete it then?

File: common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java (hunk 3 above)
Comment (Mcrtin) [reply common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java]: better readability; should i change it back?

File: common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java [hunk 5]
```diff
@@ -20,10 +20,9 @@ public enum DiscoveryType {
         this.color = color;
     }
 
-    public static DiscoveryType getDiscoveryTypeFromString(CodedString name) {
+    public static DiscoveryType getDiscoveryTypeFromString(StyledText name) {
         for (DiscoveryType type : values()) {
-            if (name.getInternalCodedStringRepresentation().charAt(1)
-                    == type.getColor().getChar()) {
+            if (name.getString().charAt(1) == type.getColor().getChar()) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java]: I am usually against changes in a port PR.. However, name.firstPart.getStyle().getColor is probably better to compare here..

File: common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java [hunk 6]
```diff
@@ -34,19 +33,18 @@
...
 
     // Test suite: https://regexr.com/778gk
     private static final Pattern POWDER_PATTERN =
-            Pattern.compile("^§7\\[(\\d+)/(\\d+)\\] Powder Slots(?: \\[§r§(.*)§r§7\\])?$");
+            Pattern.compile("^§7\\[(\\d+)/(\\d+)\\] Powder Slots(?: \\[§(.*)§7\\])?$");
 
     // Test suite: https://regexr.com/79atu
-    private static final Pattern EFFECT_LINE_PATTERN =
-            Pattern.compile("^§(.)- §r§7(.*): §r§f([+-]?\\d+)(?:§.§.)? ?(.*)$");
+    private static final Pattern EFFECT_LINE_PATTERN = Pattern.compile("^§(.)- §7(.*): §f([+-]?\\d+)(?:§.§.)? ?(.*)$");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java:47]: Have you tested that items still parse?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java:47]: (also perhaps can you update the tests?)

File: common/src/main/java/com/wynntils/utils/mc/ComponentUtils.java [hunk 7]
```diff
@@ -40,22 +41,7 @@ public static CodedString getCoded(Component component) {
...
-
-        return getCoded(component);
-    }
-
-    public static String getUnformatted(String jsonString) {
-        MutableComponent component = Component.Serializer.fromJson(jsonString);
-        if (component == null) return null;
-
-        return getUnformatted(component);
+        return StyledText.fromComponent(component).getString(PartStyle.StyleType.NONE);
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/utils/mc/ComponentUtils.java]: I am not sure if we should do this. Maybe we should just leave this be until the whole method is removed

File: common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java (hunk 3 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java]: Well, I think it is just as readable now as before. And now comes the magic sentence: "no unnecessary changes in porting PRs"

File: common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java (hunk 1 above)
Comment (Mcrtin) [reply common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java:154]: i did

File: common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java (hunk 5 above)
Comment (Mcrtin) [reply common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java]: maybe we should add a method for that 
```java
name.getFirstPart().getPartStyle().getStyle().getColor().getValue() == type.getColor().getColor()
```

File: common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java (hunk 6 above)
Comment (Mcrtin) [reply common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java:47]: i don't know how to update them but the items look the same and there are no errors except
```
[Render thread/WARN] (wynntils) Item literal{§cCharm of the Stone}[style={}] has unknown identified stat Loot from Lv. 90-110 content```

**Review by kristofbolyai — APPROVED** [review]
Looks good!
